app/models/concerns/spotlight/image_derivatives.rb

```diff
@@ -0,0 +1,59 @@
+###
+ # Include Spotlight::ImageDerivatives in a class or module to get
+ # the derivative configurations in @@spotlight_image_derivatives.
+ # A new derivative could theoretically be added by putting the
+ # following in an initializer.
+ # Spotlight::ImageDerivatives.spotlight_image_derivatives << {
+ #   version: :my_version,
+ #   field: :my_field,
+ #   lambda: lambda {|_|
+ #     version :my_version do
+ #       process :resize_to_fill => [30,30]
+ #     end
+ #   }
+ # }
+ #
+ # This will then create that new CarrierWave version in any class that extends this module
+ # and calls the apply_spotlight_image_derivative_versions class method described below.
+###
+module Spotlight
+  module ImageDerivatives
+    mattr_accessor :spotlight_image_derivatives
+
+    # Extend Spotlight::ImageDerivatives in a CarrierWave uploader
+    # then you can call this as a class method and all of the
+    # configured versions will be available
+    def apply_spotlight_image_derivative_versions
+      @@spotlight_image_derivatives.each do |version_config|
```

Comment:
Same question about using the accessor instead of the `@@`. Maybe we should even have an instance-level accessor too.